conda/utils.py

```diff
@@ -534,3 +535,13 @@ def safe_open(*args, **kwargs):
         )
 
     fp.close()
+
+
+def sanitize_filename(filename: str, replacement_char: str = "_") -> str:
+    """
+    Strips all illegal/problematic characters from a filename and replaces them
+    with `replacement_char` (defaults to "_"). We apply Windows rules
+    (see ``conda.base.constants.ILLEGAL_FILENAME_CHARS_WIN``) to err on the
+    safe side.
+    """
+    return re.sub(rf"[{ILLEGAL_FILENAME_CHARS_WIN}]", replacement_char, filename)
```

Comment:
(this is probably moot if we use a hash instead)

It's a good practice to compile regex patterns ahead of running this function so it only happens once per thread/process and not on each execution of `re.sub`.

Since the code defined in `conda/base/` is supposed to be very fast to import (according to https://github.com/conda/conda/blob/8c835dd3b54b15ed68ef1a58ba8862a0e1ddcbae/conda/base/README.md), where compiling a regex pattern may not fit the bill, I'd suggest to define this on top of the `conda/utils.py` file:

```python
ILLEGAL_FILENAME_CHARS_WIN_RE = re.compile(rf"[{ILLEGAL_FILENAME_CHARS_WIN}]")
```

And then use it below:

```suggestion
def sanitize_filename(filename: str, replacement_char: str = "_") -> str:
    """
    Strips all illegal/problematic characters from a filename and replaces them
    with `replacement_char` (defaults to "_"). We apply Windows rules
    (see ``conda.base.constants.ILLEGAL_FILENAME_CHARS_WIN``) to err on the
    safe side.
    """
    return ILLEGAL_FILENAME_CHARS_WIN_RE.sub(replacement_char, filename)
```